src/main/java/net/apnic/whowas/App.java

```diff
@@ -112,15 +98,19 @@ public TaskExecutor taskScheduler() {
     ApplicationContext context;
 
     @Autowired
-    private JdbcOperations jdbcOperations;
-
     private RipeDbLoader dbLoader;
+
     private Exception lastDbException = null;
 
     @PostConstruct
-    public void initialise() {
-        dbLoader = new RipeDbLoader(jdbcOperations, -1L);
-        executorService.execute(this::buildTree);
+    public void initialise() throws ExecutionException, InterruptedException {
+        /*
+          Scoping the initial load of data to @PostConstruct
+          results in the loading completing before the servlet context is started,
+          preventing requests from being served from partially loaded data.
+        */
+        Future initialDataLoaded = executorService.submit(this::buildTree);
+        initialDataLoaded.get();
```

Comment:
The idea of blocking until loading has happened is interesting but I feel we are solving a problem that is not for the app to solve on its own. Perhaps the better way to tackle this is to implement the health endpoint for spring so that K8 deploys can be configured to use this.

Seems like a better solution for the app to be queryable as to state.

Reply:
Not opening the socket until the service can serve queries is a simple approach that makes it impossible to miss-configure a deployment such that responses may be served from a partially loaded data set.